kapitan/refs/secrets/vault.py

```diff
@@ -0,0 +1,167 @@
+# Copyright 2019 The Kapitan Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"hashicorp vault secrets module"
+
+import hvac
+import base64
+from hvac.exceptions import Forbidden, VaultError
+from yaml import safe_load
+from os.path import join,expanduser
+from os import getenv
+from sys import argv
+
+from kapitan.refs.base import Ref, RefBackend, RefError
+from kapitan import cached
+from kapitan.errors import KapitanError
+
+class VaultError(KapitanError):
+    """Generic vault errors"""
+    pass
+
+def get_env():
```

Comment:
I'd advise you not to only support env variables - ensuring the values are in the inventory is more kapitanesque :)
I'd like to see this supporting vault settings in both env and inventory.

Reply:
The reason I didn't use inventory as a store for the variable is due to the security concerns, we would be revealing the server address or tokens, but instead of those all other variables I guess can be included. Would that be acceptable?

Reply:
Support for both (env variables & values from inventory) are added now